### PR TITLE
Fix Wconversion issues for clang

### PR DESCRIPTION
**Pull request by nakarlsson**
Add explicit uint16_t cast for EoE header SET macros. Used for SET of FrameInfo 1 and FrameInfo 2.

Increase datatype for offset used by EOE_HDR_FRAME_OFFSET_GET.

Change debug print formatting to %u for unsigned datatypes.

### CODE REVIEW

**Review by hefloryd — APPROVED** [review]

**Comment by nakarlsson** [comment]
@ArthurKetels , I'd like to get an OK from you as well before adding the use of <inttype.h>

**Comment by nakarlsson** [comment]
Ping @ArthurKetels

File: soes/esc_coe.c [hunk 1]
```diff
@@ -299,7 +299,7 @@ static void SDO_upload (void)
    uint16_t index;
    uint8_t subindex;
    int32_t nidx;
-   int32_t nsub;
+   int16_t nsub;
```
Comment (ArthurKetels) [inline soes/esc_coe.c:305]: Why is this changed back to int16_t?

**Comment by ArthurKetels** [comment]
Although I really hate the readability of the <inttype.h> change it is an acceptable compromise. A better alternative would have been an upgraded printf that just writes the correct data type instead of making subtle changes.

For me clearly written code prevents more bugs than chasing pedantic compiler warnings. We are coding in C, an old language with warts and spots, not Rust or something like that.

Other than that, please make it so.

File: soes/esc_coe.c (hunk 1 above)
Comment (nakarlsson) [reply soes/esc_coe.c:305]: I think it was unnecessarily promoted to int32_t in last commit, in most cases nsub is int16_t trough out the file and can be it in SDO_upload as well,. So, the idea was to align the usage in this context and I changed it back.
No warning.

**Comment by nakarlsson** [comment]
>For me clearly written code prevents more bugs than chasing pedantic compiler warnings. We are coding in C, an old language with warts and spots, not Rust or something like that.

Point taken.
